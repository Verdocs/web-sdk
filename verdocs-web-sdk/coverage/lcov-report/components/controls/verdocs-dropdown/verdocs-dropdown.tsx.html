
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for components/controls/verdocs-dropdown/verdocs-dropdown.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">components/controls/verdocs-dropdown</a> verdocs-dropdown.tsx</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">60% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>12/20</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>5/10</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">37.5% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>3/8</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">60% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>12/20</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {Host, State} from '@stencil/core';
import {randomString} from '@verdocs/js-sdk';
import {Component, Prop, Element, h, Event, EventEmitter} from '@stencil/core';
&nbsp;
const DropdownArrow = `&lt;svg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="#50BE80"&gt;&lt;path d="M0 0h24v24H0V0z" fill="none"/&gt;&lt;path d="M7 10l5 5 5-5H7z"/&gt;&lt;/svg&gt;`;
&nbsp;
export interface IMenuOption {
  label: string;
  id?: any;
  faIcon?: any;
  disabled?: boolean;
}
&nbsp;
/**
 * Display a drop-down menu button. A menu of the specified options will be displayed when the button is pressed. The menu will be hidden
 * when the button is pressed again, or an option is selected. Separators may be created by supplying an entry with an empty label.
 *
 * ```ts
 * &lt;verdocs-dropdown
 *   options={[
 *     {label: 'Option 1', disabled: true},
 *     {label: 'Option 2', id: '2'}
 *     {label: ''}
 *     {label: 'Option 3', id: '2'}
 *    ]}
 *   label="OK" onClick={() =&gt; (console.log('OK clicked'))}
 * /&gt;
 * ```
 */
@Component({
  tag: 'verdocs-dropdown',
  styleUrl: 'verdocs-dropdown.scss',
})
export class VerdocsDropdown {
  @Element()
  el: HTMLElement;
&nbsp;
  /**
   * The menu options to display.
   */
  @Prop() options: IMenuOption[] = [];
&nbsp;
  /**
   * Event fired when a menu option is clicked.
   * Web Component events need to be "composed" to cross the Shadow DOM and be received by parent frameworks.
   */
  @Event({composed: true}) optionSelected: EventEmitter&lt;IMenuOption&gt;;
&nbsp;
  @State() containerId = `verdocs-send-${Math.random().toString(36).substring(2, 11)}`;
&nbsp;
  @State() showPicker = false;
&nbsp;
<span class="fstat-no" title="function not covered" >  handleSelectOption(</span>e: any, option: IMenuOption) {
<span class="cstat-no" title="statement not covered" >    e.stopPropagation();</span>
<span class="cstat-no" title="statement not covered" >    this.optionSelected?.emit(option);</span>
<span class="cstat-no" title="statement not covered" >    this.showPicker = false;</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  toggle(</span>) {
<span class="cstat-no" title="statement not covered" >    this.showPicker = !this.showPicker;</span>
  }
&nbsp;
  render() {
    const elId = `verdocs-dropdown-${randomString(8)}`;
&nbsp;
    return (
      &lt;Host&gt;
        &lt;button class="arrow" innerHTML={DropdownArrow} aria-label="Open Menu" id={elId} onClick={<span class="fstat-no" title="function not covered" >() =</span>&gt; (<span class="cstat-no" title="statement not covered" >this.showPicker = !this.showPicker)</span>} /&gt;
&nbsp;
        {this.showPicker &amp;&amp; (
          &lt;verdocs-portal anchor={elId} onClickAway={<span class="fstat-no" title="function not covered" >() =</span>&gt; (<span class="cstat-no" title="statement not covered" >this.showPicker = false)</span>} id="verdocs-dropdown-menu-items"&gt;
            {this.options.map(option =&gt; {
              <span class="missing-if-branch" title="if path not taken" >I</span>if (!option.label) {
<span class="cstat-no" title="statement not covered" >                return &lt;div class="verdocs-dropdown-menu-separator" /&gt;;</span>
              } else {
                return (
                  &lt;button class="verdocs-dropdown-menu-option" disabled={option.disabled || false} onClick={<span class="fstat-no" title="function not covered" >e </span>=&gt; <span class="cstat-no" title="statement not covered" >this.handleSelectOption(e, option)}</span>&gt;
                    {option.label}
                  &lt;/button&gt;
                );
              }
            })}
          &lt;/verdocs-portal&gt;
        )}
      &lt;/Host&gt;
    );
  }
}
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2025-07-24T13:19:16.585Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    