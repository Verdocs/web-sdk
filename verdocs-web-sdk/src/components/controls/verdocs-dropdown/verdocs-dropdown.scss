verdocs-dropdown {
  /* Fallback font for old browsers */
  font-family: sans-serif;
  font-family: var(--verdocs-font);
  display: block;

  .arrow {
    width: 32px;
    height: 26px;
    background: #fff;
    cursor: pointer;
    border-radius: 4px;
    align-items: center;
    display: inline-flex;
    border: 1px solid #ccc;
    justify-content: center;
  }

  .arrow svg {
    fill: var(--verdocs-primary-color);
    transition: 0.3s;
  }

  .items {
    width: 200px;
    display: none;
    z-index: 10000;
    border-radius: 5px;
    background: #33364b;
    border: 1px solid #654dcb;
  }

  .items[data-show] {
    display: block;
  }

  .option {
    width: 100%;
    color: #fff;
    border: none;
    display: block;
    text-align: left;
    background: none;
    padding: 10px 20px;
  }

  .option:hover {
    cursor: pointer;
    background: var(--verdocs-primary-color);
  }

  .option[disabled] {
    color: #aaa;
  }

  .option[disabled]:hover {
    color: #aaa;
    cursor: inherit;
    background: transparent;
  }

  &.open {
    .items {
      display: block;
    }

    .arrow {
      background: #4c56cb;
      border: 1px solid #707ae5;
    }

    .arrow svg {
      /*transform: rotate(180deg);*/
      transform: scaleY(-1);
      fill: #fff;
    }
  }
}
