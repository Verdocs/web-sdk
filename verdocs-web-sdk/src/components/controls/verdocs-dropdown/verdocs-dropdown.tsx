import {Host} from '@stencil/core';
import {createPopper, Instance} from '@popperjs/core';
import {Component, Prop, State, h, Event, EventEmitter} from '@stencil/core';
import SortDown from './down-arrow.svg';

export interface IMenuOption {
  label: string;
  id?: any;
  faIcon?: any;
  disabled?: boolean;
}

/**
 * Display a drop-down menu button. A menu of the specified options will be displayed when the button is pressed. The menu will be hidden
 * when the button is pressed again, or an option is selected.
 */
@Component({
  tag: 'verdocs-dropdown',
  styleUrl: 'verdocs-dropdown.scss',
})
export class VerdocsDropdown {
  private dropdownButton?: HTMLButtonElement;
  private dropdownMenu?: HTMLDivElement;
  private popper?: Instance;

  /**
   * The menu options to display.
   */
  @Prop() options: IMenuOption[] = [];

  /**
   * If set, the component will be open by default. This is primarily intended to be used for testing.
   */
  @Prop() open: boolean;

  /**
   * If set, the component will be open by default.
   */
  @State() isOpen: boolean;

  /**
   * Event fired when a menu option is clicked.
   * Web Component events need to be "composed" to cross the Shadow DOM and be received by parent frameworks.
   */
  @Event({composed: true}) optionSelected: EventEmitter<IMenuOption>;

  componentWillLoad() {
    this.isOpen = !!this.open;
  }

  componentDidLoad() {
    this.popper = createPopper(this.dropdownButton, this.dropdownMenu, {placement: 'bottom-start', modifiers: [{name: 'offset', options: {offset: [0, 6]}}]});
  }

  handleSelectOption(e: any, option: IMenuOption) {
    e.stopPropagation();
    this.isOpen = false;
    this.optionSelected.emit(option);
    this.hide();
  }

  // See https://popper.js.org/docs/v2/tutorial/
  // What we're doing here is clearing event listeners when they aren't needed, to increase performance in lists
  showDropdown() {
    this.isOpen = true;
    this.dropdownMenu.setAttribute('data-show', '');
    this.dropdownMenu.removeAttribute('aria-hidden');

    this.popper?.setOptions(options => ({
      ...options,
      modifiers: [...options.modifiers, {name: 'eventListeners', enabled: true}],
    }));

    this.popper?.update();
  }

  handleHideDropdown(e: any) {
    if (e?.target?.localName === 'button' && e?.target?.className === 'arrow') {
      // This event is fired when a menu element is clicked because the button "loses focus" then too
      return;
    }

    this.hide();
  }

  toggleDropdown(e: any) {
    e.stopPropagation();

    if (this.isOpen) {
      this.hide();
    } else {
      this.showDropdown();
    }
  }

  hide() {
    this.isOpen = false;
    this.dropdownMenu.removeAttribute('data-show');
    this.dropdownMenu.setAttribute('aria-hidden', 'true');
    this.popper?.setOptions(options => ({
      ...options,
      modifiers: [...options.modifiers, {name: 'eventListeners', enabled: false}],
    }));
  }

  render() {
    return (
      <Host class={{storybook: !!window?.['STORYBOOK_ENV'], open: !!this.isOpen}}>
        <button
          class="arrow"
          innerHTML={SortDown}
          aria-label="Open Menu"
          onClick={e => this.toggleDropdown(e)}
          onBlur={e => this.handleHideDropdown(e)}
          ref={el => (this.dropdownButton = el as HTMLButtonElement)}
        />

        <div class="items" aria-hidden={!this.open} ref={el => (this.dropdownMenu = el as HTMLDivElement)}>
          {this.options?.map(option => (
            <button onClick={e => this.handleSelectOption(e, option)} class="option" disabled={option.disabled}>
              {option.label}
            </button>
          ))}
        </div>
      </Host>
    );
  }
}
