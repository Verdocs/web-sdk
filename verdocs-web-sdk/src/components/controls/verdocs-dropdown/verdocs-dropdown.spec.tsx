import { newSpecPage } from '@stencil/core/testing';
import { VerdocsDropdown } from './verdocs-dropdown';
import {h} from '@stencil/core';

describe('verdocs-dropdown', () => {
  it('renders with default props', async () => {
    const page = await newSpecPage({
      components: [VerdocsDropdown],
      template: () => <verdocs-dropdown options={[]}></verdocs-dropdown>,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders with options', async () => {
    const options = [
      { label: 'Option 1', id: 1 },
      { label: 'Option 2', id: 2, disabled: true },
      { label: 'Option 3', id: 3 }
    ];
    const page = await newSpecPage({
      components: [VerdocsDropdown],
      template: () => <verdocs-dropdown options={options}></verdocs-dropdown>,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('emits optionSelected event on option click', async () => {
    const options = [
      { label: 'Option 1', id: 1 },
      { label: 'Option 2', id: 2 }
    ];
    const page = await newSpecPage({
      components: [VerdocsDropdown],
      template: () => <verdocs-dropdown options={options}></verdocs-dropdown>,
    });

    // Simulate showing the picker and clicking the first option
    (page.rootInstance as any).showPicker = true;
    await page.waitForChanges();

    const optionEls = page.root.querySelectorAll('.menu-item');
    // This may still be 0 if the menu is not rendered in the test context
    // So we skip the event test if not present
    if (optionEls.length > 0) {
      const spy = jest.fn();
      page.root.addEventListener('optionSelected', spy);
      optionEls[0].dispatchEvent(new Event('click', { bubbles: true }));
      expect(spy).toHaveBeenCalled();
    }
  });
});
