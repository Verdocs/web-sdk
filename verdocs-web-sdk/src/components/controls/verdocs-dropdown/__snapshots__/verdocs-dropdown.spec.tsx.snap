// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verdocs-dropdown renders with default props 1`] = `
<verdocs-dropdown>
  <button aria-label="Open Menu" class="arrow" id="verdocs-dropdown-4fzzzxjyl">
    <svg fill="#50BE80" height="24px" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
      <path d="M0 0h24v24H0V0z" fill="none"></path>
      <path d="M7 10l5 5 5-5H7z"></path>
    </svg>
  </button>
</verdocs-dropdown>
`;

exports[`verdocs-dropdown renders with options 1`] = `
<verdocs-dropdown>
  <button aria-label="Open Menu" class="arrow" id="verdocs-dropdown-4fzzzxjyl">
    <svg fill="#50BE80" height="24px" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
      <path d="M0 0h24v24H0V0z" fill="none"></path>
      <path d="M7 10l5 5 5-5H7z"></path>
    </svg>
  </button>
</verdocs-dropdown>
`;
